src/bun.js/node/win_watcher.zig

```diff
@@ -0,0 +1,387 @@
+        if (this.file_paths.getEntry(file_path)) |entry| {
+            var path = entry.value_ptr;
+            if (path.refs > 0) {
+                path.refs -= 1;
+                if (path.refs == 0) {
+                    const path_ = path.path;
+                    this.main_watcher.remove(path.hash);
+                    _ = this.file_paths.remove(path_);
+                    bun.default_allocator.free(path_);
+                }
+            }
+        }
+    }
+
+    // unregister is always called form main thread
+    fn unregisterWatcher(this: *WinPathWatcherManager, watcher: *PathWatcher) void {
+        var watchers = this.watchers.slice();
+        defer {
+            if (this.deinit_on_last_watcher and this.watcher_count == 0) {
+                this.deinit();
+            }
+        }
+
+        for (watchers, 0..) |w, i| {
+            if (w) |item| {
+                if (item == watcher) {
+                    watchers[i] = null;
+                    // if is the last one just pop
+                    if (i == watchers.len - 1) {
+                        this.watchers.len -= 1;
+                    }
+                    this.watcher_count -= 1;
+
+                    this._decrementPathRef(watcher.path.path);
+                    break;
+                }
+            }
+        }
+    }
+    fn deinit(this: *WinPathWatcherManager) void {
+        // enable to create a new manager
+        if (default_manager == this) {
+            default_manager = null;
+        }
+
+        // only deinit if no watchers are registered
+        if (this.watcher_count > 0) {
+            // wait last watcher to close
+            this.deinit_on_last_watcher = true;
+            return;
+        }
+
+        this.main_watcher.deinit(false);
+
+        if (this.watcher_count > 0) {
+            while (this.watchers.popOrNull()) |watcher| {
+                if (watcher) |w| {
+                    // unlink watcher
+                    w.manager = null;
+                }
+            }
+        }
+
+        // close all file descriptors and free paths
+        var it = this.file_paths.iterator();
+        while (it.next()) |*entry| {
+            const path = entry.value_ptr.*;
+            _ = bun.sys.close(path.fd);
+            bun.default_allocator.free(path.path);
+        }
+
+        this.file_paths.deinit();
+
+        this.watchers.deinitWithAllocator(bun.default_allocator);
+
+        this.destroy();
+    }
+};
+
+pub const PathWatcher = struct {
+    handle: uv.uv_fs_event_t,
+    ctx: ?*anyopaque,
+    recursive: bool,
+    callback: Callback,
+    flushCallback: UpdateEndCallback,
+    manager: ?*WinPathWatcherManager,
+    path: WinPathWatcherManager.PathInfo,
+    last_change_event: ChangeEvent = .{},
+    closed: bool = false,
+    needs_flush: bool = false,
+
+    pub usingnamespace bun.New(PathWatcher);
+
+    const log = Output.scoped(.WinPathWatcher, false);
```

Comment:
only one PathWatcher per platform so the Win prefix is unnecessary

```suggestion
    const log = Output.scoped(.PathWatcher, false);
```